mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -1292,10 +1293,6 @@ struct GridwiseGemmV2RewritePattern
           arrayA, arrayB, regCAllocOp, op.getBlockSizeAttr(),
           op.getParamsAttr());
 
-      // LDS barrier.
-      // This barrier prevents halo part of outputs having weird values.
-      b.create<LDSBarrierOp>(loc);
```

Comment:
Yes, this change sets barrier after all waves got their data in the register to prevent that.

Reply:
Ah ... yeah, that's actually a good point. Can you comment that over in gridwise gemm that the barrier is in blockwise gemm?

Reply:
I'll be happy to land this, thanks for the explanation.

Reply:
(but can you do this for non-xdlops too?)

Reply:
I'm not sure this also applies to non-xdlops, need to have a look into non-xdlops loop.

Reply:
It's got the same high-level structure.

Also, can we get a by-hand perf run here?

Reply:
I'll have a look

Reply:
Unfortunately perf run proved this is not better. Mostly par but quite a few kernels show ~1% slower. 
Possibly because moving barrier before the mfma forced mfma to wait for all waves to finish reading from LDS and that's more than the benefits from the mfma overlapping with the later instructions.

Also experimented with simply removing that second LDS barrier, also doesn't give a better overall performance - 2 configs got 10% faster but 3 configs got 2 times slower.

Will close this, no point to have it. Was nice to refresh my memory around this.

Reply:
Thanks for experimenting!

Reply:
Yes, I think the both cases lds barrier guarantees LDS usage are done with current iteration before moving to the next. 

It is easier to maintain/understand this with both barriers emitted in the same lowering pass.